docs/section5/Rule5-8.md

```diff
@@ -0,0 +1,45 @@
+
+# Envelope - Rule 5-8  
+
+**Rule ID:** 5-8  
+**Rule Description:**  Fenestration U-factors for new buildings, existing buildings, and additions shall match the appropriate requirements in Tables G3.4-1 through G3.4-8 for the applicable glazing percentage for Uall.  
+**Rule Assertion:** Baseline RMR = expected value  
+**Appendix G Section:** Section G3.1-5(d) Building Envelope Modeling Requirements for the Baseline building  
+**Appendix G Section Reference:** Tables G3.4-1 to G3.4-8  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:**  None  
+
+**Manual Check:** None  
+**Evaluation Context:** Each Data Element  
+**Data Lookup:** Tables G3.4-1 to G3.4-8  
+
+## Rule Logic:  
+
+- Get building climate zone: ```climate_zone = B_RMR.weather.climate_zone```  
+
+- For each building segment in the Baseline model: ```for building_segment_baseline in B_RMR.building.building_segments:```  
+
+  - For each thermal_block in building segment: ```for thermal_block_baseline in building_segment_baseline.thermal_blocks:```  
+
+  - For each zone in thermal block: ```for zone_baseline in thermal_block_baseline.zones:```  
+
+  - For each space in thermal zone: ```for space_baseline in zone_baseline.spaces:```  
+
+    - Get surfaces in space: ```for surface_baseline in space_baseline.surfaces:```  
+
+      - Check if surface is exterior wall: ```if ( surface_baseline.classification == "WALL" ) AND ( surface_baseline.adjacent_to == "AMBIENT" ): wall_surface_baseline = surface_baseline```  
+
+        - Calculate the total wall area: ```total_wall_area_baseline += wall_surface_baseline.area```  
+
+        - Get the U-factor for vertical glazings: ```window_performance_value_baseline.append(window.u_factor for window in wall_surface_baseline.fenestration_subsurfaces)```  
+
+        - Calculate the total vertical glazing area: ```total_window_area_baseline += sum(window.area for window in wall_surface_baseline.fenestration_subsurfaces)```  
+
+        - Calculate the window to wall ratio: ```window_wall_ratio_baseline = total_window_area_baseline / total_wall_area_baseline```  (Note XC, can we get the WWR and skylight roof ratio from Rule 5-7 instead of calculating it here?)
+
+    - Get space conditioning type: ```space_conditioning_type_baseline = space_baseline.conditioning_type```  
+
```

Comment:
There needs to be a step to determine is the space is residential or nonresidential. Residential space types include guest rooms, living quarters, private living space, and sleeping quarters. This would bee determined based on the lighting space type.

Reply:
Thanks Supriya, I missed the residential part of the table. Just want to reference back to the discussion on residential vs non-residential for lighting. We are using Table G3.7 for the lighting space types. I found four space spaces that are residential, Dormitory - Living Quarters, Fire Station - Sleeping Quarters, Guest Room and Dwelling Unit. So all envelope in conditioned spaces under these space types use the "residential" column in Table G3.4.  All semi-heated spaces (residential and non-residential) use the "semi-heated" column. @JasonGlazer @supriyagoel @mkarpman Please confirm.

Reply:
@xingcx, residential spaces should also include healthcare facilities patient rooms, perhaps also nursery.
Also, I believe App G requires use of space-by-space method to determine baseline lighting (Tabld G3.1 #6:  Interior lighting power in the baseline building design shall be determined using the values in Table G3.7.). Building area method may only be used for verifying compliance with G1.2.1(b).

Reply:
@xingcx - Maria is right. I pulled the list of residential spaces from the wrong location, Residential spaces include Dormitory - Living Quarters, Fire Station - Sleeping Quarters, Guest Room and Dwelling Unit, hospital patient rooms and nursery (in a hospital).
@mkarpman I will copy your comment on lighting to the appropriate lighting RDS and tag you there for a discussion.